### PR TITLE
CLN: simplify info

### PR DESCRIPTION
**Pull request by MarcoGorelli**
precursor to #31796

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/io/formats/info.py [hunk 1]
```diff
@@ -228,14 +229,14 @@ def _verbose_repr():
             + _put_str("-" * len_dtype, space_dtype)
         )
 
-        for i, col in enumerate(data.columns):
-            dtype = data.dtypes.iloc[i]
+        for i, col in enumerate(cols):
+            dtype = dtypes[i]
```
Comment (mroeschke) [inline pandas/io/formats/info.py:233]: Is there a particular reason to use `__getitem__` instead of `iloc` here? (same with `counts` below)

Comment (MarcoGorelli) [reply pandas/io/formats/info.py:233]: Because in #31796, in the case that `data` is a Series, I make `counts` an `Index` (which has no attribute `.iloc`)

Comment (mroeschke) [reply pandas/io/formats/info.py:233]: I see. Thanks for the explanation

**Comment by mroeschke** [comment]
Thanks @MarcoGorelli

**Pull request merged**